bin/ddg2logo

```diff
@@ -60,169 +55,53 @@ DESTABILIZING MUTATIONS
-                    if pos_dic[position]==config[key]['color']:
-                        tmp.append(position)
-                color_list.append(tmp)
-            
-            #initialize list sorted by color and list with residues 
-            #not in the config
-            sorted_list=[]
-            no_config_res=[]
-
-            #color_list flattend 
-            flat_list=[item for sublist in color_list for item in sublist]
-
-            #parse the s_pos_idx and append entries if found in the 
-            #color_list
-            for lista in color_list:
-                tmp=[]
-
-                #check if residues are not in the config and add them to
-                #list of residues not in the config only once
-                for i in s_pos_idx:
-                    if i in lista:
-                        tmp.append(i)
-                    else:
-                        if i not in flat_list:
-                            no_config_res.append(i)
-                #if the list is not empty, add it to the sorted list
-                if tmp:
-                    sorted_list.append(tmp)
-
-            #append list of unique residues not in the config 
-            sorted_list.append(list(set(no_config_res)))
-
-            #remove empty lists
-            for lista in sorted_list:
-                if not lista:
-                    sorted_list.remove(lista)
-
-            #create numpy array from the sorted list 
-            if sorted_list:
-                s_pos_idx = np.hstack(sorted_list)
-
-        # if too many of them are allowed, use X and continue
-        if len(s_pos_idx) >= options.degeneracy:
-            ax.imshow(letters['X'],
-                      extent=(hgap + x*width + x*hgap, hgap + (x+1)*width + x*hgap, 0.0, max_y/4.0),
-                      aspect='auto')
-
-            continue
-
-        # initialize y value accumulator
-        accumulator = 0
-
-        # for every DDG value
-        for idx, i in enumerate(s_pos_idx):
-
-            # add vertical gap (unless it's the first letter)
-            if idx == 0:
-                low_y = accumulator
-            else:
-                low_y = accumulator + vgap
-
-            # add DDG
-            accumulator += c[i]
-
-            # plot
-            ax.imshow(  letters[reslist[i]],
-                        extent=(hgap + x*width + x*hgap, hgap + (x+1)*width + x*hgap, low_y, accumulator),
-                        aspect='auto' )
-
-    ax.set_ylabel(r'modified FoldX $\Delta\Delta{}$G (kcal/mol)')
-
-    ax.set_xlim((0, mat.shape[0]*width +hgap*mat.shape[0] + hgap))
-    ax.set_ylim((0, max_y))
-
-    ax.set_xticks(xticks)
-    ax.set_xticklabels(xlabels)
-
-    plt.setp(ax.get_xticklabels(), size=fontsize, rotation=90)
-    plt.tight_layout()
+    # style using Logo methods
+    fig.style_spines(visible=False)
+    fig.style_spines(spines=['left', 'bottom'], visible=True)
+    fig.style_xticks(rotation=45, fmt='%d', anchor=0)
```

Comment:
i would choose 90 as a standard, good for multimers.

Reply:
Agree I will change it